### PR TITLE
add contact list creation and tests

### PR DESCRIPTION
**Pull request by bryanmontz**
Adds the ability to create contact lists from a list of pubkeys or pubkey tags. I provided both options so it can be very simple if the client prefers, but it also has full functionality matching the spec (for petnames).

### CODE REVIEW

File: Sources/NostrSDK/Events/ContactListEvent.swift [hunk 1]
```diff
@@ -32,6 +32,19 @@ public struct RelayPermissions: Equatable {
 /// > Note: [NIP-02 Specification](https://github.com/nostr-protocol/nips/blob/master/02.md#contact-list-and-petnames)
 public final class ContactListEvent: NostrEvent {
     
+    public required init(from decoder: Decoder) throws {
+        try super.init(from: decoder)
+    }
+    
+    @available(*, unavailable, message: "This initializer is unavailable for this class.")
+    override init(kind: EventKind, content: String, tags: [Tag] = [], createdAt: Int64 = Int64(Date.now.timeIntervalSince1970), signedBy keypair: Keypair) throws {
```
Comment (bryanmontz) [inline Sources/NostrSDK/Events/ContactListEvent.swift:40]: I'm trying something fairly opinionated here, so please let me know your feedback. I'm marking the parent class's initializer as unavailable for this class. That way I can prevent client apps from creating an event like this, where the class and the kind don't match:
```swift
let badEvent = ContactListEvent(kind: .recommendServer, tags:...
```

Another great thing about this is that code completion won't even suggest that initializer.

Comment (joelklabo) [reply Sources/NostrSDK/Events/ContactListEvent.swift:40]: Yeah, I think that's a good idea. You can always use the `NostrEvent` initializer if needed for something custom.

**Review by joelklabo — APPROVED** [review]
Only suggestion is to verify the note in the test. Only because that's what we've done for the others. But, I could definitely be persuaded that it isn't necessary since you're not adding anything that has an effect on verification.

File: Tests/NostrSDKTests/EventCreatingTests.swift [hunk 2]
```diff
@@ -81,6 +61,58 @@ final class EventCreatingTests: XCTestCase, EventCreating, EventVerifying {
...
+        let tags: [Tag] = [
+            Tag(name: .pubkey, value: "83y9iuhw9u0t8thw8w80u", otherParameters: ["bob"]),
+            Tag(name: .pubkey, value: "19048ut34h23y89jio3r8", otherParameters: ["alice"]),
+            Tag(name: .pubkey, value: "5r623gyewfbh8uuiq83rd", otherParameters: ["steve"])
+        ]
+        
+        let event = try contactList(withPubkeyTags: tags,
+                                    signedBy: Keypair.test)
+        
+        XCTAssertEqual(event.tags, tags)
```
[the remark this answers is not in the document]
Comment (bryanmontz) [reply Tests/NostrSDKTests/EventCreatingTests.swift:96]: I just forgot. We should always do that. Thanks!

**Pull request merged**